src/orion/algo/hyperband.py

```diff
@@ -557,16 +639,23 @@ def get_candidates(self, rung_id):
         if self.has_rung_filled(rung_id + 1):
             return []
 
-        rung = self.rungs[rung_id]["results"]
+        rung_results = self.rungs[rung_id]["results"]
         next_rung = self.rungs[rung_id + 1]["results"]
+        # BUG: What if some of the objectives are None? Then comparison between None and floats here
```

Comment:
I think we should filter them to only select objectives that are not None in `rung_results`. We can then sort safely.

Reply:
Again, this is out-of-scope IMO. 
This assertion here is making the implicit error more explicit. It'll be the job of another PR to debug/improve these algos as needed.

Reply:
Also, more generally, in this kind of case, I think leaving a "BUG" comment is enough, but if you want me to make an issue to keep track of this, I can also make one.

Reply:
It's already difficult to keep track of all the issues. Leaving BUG comments is to likely to be missed for a long time in my opinion. I can create the issues this time as we discuss. Next time feel free to create the issues as you find bugs like these.

Regarding the assert, it's better to use explicit error messages. The users will be confused if they get asserts like this and won't understand what is happening. We can at least add a message to the assert, but the type of error message itself always convey information.

I created #858

Reply:
Fixed in 2c1f8c1dc08715e867ffa436e97245fc083630e7

Reply:
Alright, and for assert we leave it like that and defer the discussion here: https://github.com/Epistimio/orion/issues/859